blockwork/config/parser.py

```diff
@@ -12,37 +12,105 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import contextlib
 from pathlib import Path
-from typing import TypeVar
+from typing import TypeVar, cast
 
 import yaml
 from blockwork.common.yaml import Parser, DataclassConverter
+from blockwork.context import Context
 
 from . import base, registry
 
-_Convertable = TypeVar('_Convertable')
-
 
 class Site(Parser):
     "Parser for site yaml files"
-    def __init__(self):
+    def __init__(self, ctx: Context):
         super().__init__(registry.site)
+        self.ctx = ctx
 
 class Project(Parser):
     "Parser for project yaml files"
-    def __init__(self, site: base.Site):
+    def __init__(self, ctx: Context, site: base.Site):
         super().__init__(registry.project)
+        self.ctx = ctx
         self.site = site
 
 class Element(Parser):
     "Parser for 'element' yaml files where an 'element' is a unit of configuration within the target" 
-    def __init__(self, site: base.Site, project: base.Project):
+    def __init__(self, ctx: Context, site: base.Site, project: base.Project):
         super().__init__(registry.element)
+        self.ctx = ctx
         self.site = site
         self.project = project
+        self._block_stack: list[str] = []
+
+    def parse_target(self, target_spec: str, target_type: type[base.Element]):
+        """
+        Parse a target config file based on the target block, path within that 
+        block and expected type
+
+        :param target_spec: The target specified as `<block>.<path>` or `<path>` 
+                            where path is the path within the block to the yaml,
+                            not including the extension. The <block> component
+                            is inferred where possible.
+        :param target_type: Expected element type
+        """
+        target_parts = target_spec.split('.')
+        if len(target_parts) == 2:
```

Comment:
Do we only ever expect there to be one or two path segments? How does something like `tech.cell.xyz` get handled?

Reply:
Currently it's based purely on the "dumb" mapping in the project yaml.

example/projects/example.yaml

so if you had an entry `tech/cell: <path>` in there you would specify it as `tech/cell.xyz`